app/main.py

```diff
@@ -1,34 +1,63 @@
+from __future__ import annotations
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple[int, int],
+            end: tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+
+        self.is_drowned = is_drowned
+        self.start = start
+        self.end = end
+        self.decks = []
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        crushed_deck = self.get_deck(row, column)
+        if crushed_deck is not None:
+            crushed_deck.is_alive = False
+
+        if crushed_deck in self.decks:
+            if not any(deck.is_alive for deck in self.decks):
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple[tuple]]) -> None:
+        self.fields = {}
+        for ship in ships:
+            self.ship = Ship(ship[0], ship[1])
```

Comment:
```suggestion
            ship = Ship(ship[0], ship[1])
```
I believe you don’t need ship as an attribute of instance